onnxruntime/core/providers/webnn/builders/impl/qdq_op_builder.cc

```diff
@@ -42,70 +42,28 @@ Status QDQOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   int32_t zero_point_type = 0;
   ORT_RETURN_IF_NOT(GetType(*input_defs[0], input_type, logger), "Cannot get input data type");
   ORT_RETURN_IF_NOT(GetType(*output_defs[0], output_type, logger), "Cannot get output data type");
+
   emscripten::val input = model_builder.GetOperand(input_defs[0]->Name());
   emscripten::val scale = model_builder.GetOperand(input_defs[1]->Name());
-
   emscripten::val zero_point = emscripten::val::null();
+
   if (input_defs.size() == 3 && input_defs[2]->Exists()) {
     zero_point = model_builder.GetOperand(node.InputDefs()[2]->Name());
   } else {
     // DequantizeLinear: x_zero_point's data type equals to input data type
     // QuantizeLinear: x_zero_point's data type equals to output data type
+    // WebNN requires the zero_point to have the same shape as the scale
```

Comment:
Is this shape requirement notable to the WebNN EP? For blocked scale&zp, ONNX itself also requires that:

```
x_scale (heterogeneous) - T2:
...
x_zero_point (optional, heterogeneous) - T1:
... Shape must match x_scale ...
```

https://onnx.ai/onnx/operators/onnx__DequantizeLinear.html

Reply:
Yes, this requirement is for WebNN EP. This time `zero_point` is not provided (optional) by ONNX model, however WebNN requires the `zero_point`, we need to create a `zero_point` constant in the WebNN EP.

Reply:
Ah, so that's what the comment really means.